In [1]:
import pandas as pd

In [3]:
import numpy as np

In [7]:
data=pd.read_csv('C:/Users/malte/OneDrive/Desktop/Sujay/advertising.csv')

In [9]:
data.head()

TV  Radio  Newspaper  Sales
0  230.1   37.8       69.2   22.1
1   44.5   39.3       45.1   10.4
2   17.2   45.9       69.3   12.0
3  151.5   41.3       58.5   16.5
4  180.8   10.8       58.4   17.9

In [11]:
import statsmodels.api as sm

In [13]:
X = data[['TV', 'Radio', 'Newspaper']]

In [15]:
y = data['Sales']

In [17]:
X = sm.add_constant(X)

In [19]:
model = sm.OLS(y, X).fit()

In [21]:
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                  Sales   R-squared:                       0.903
Model:                            OLS   Adj. R-squared:                  0.901
Method:                 Least Squares   F-statistic:                     605.4
Date:                Mon, 10 Nov 2025   Prob (F-statistic):           8.13e-99
Time:                        22:07:57   Log-Likelihood:                -383.34
No. Observations:                 200   AIC:                             774.7
Df Residuals:                     196   BIC:                             787.9
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          4.6251      0.308     15.041      0.0

90.3% of the variation in Sales is explained by TV, Radio, and Newspaper ads. Very strong fit.

Adjusted for number of predictors. Still very high → model is not overfitting.

The overall model is statistically significant.